### PR TITLE
First cut of Pathname#relative_path_from

### PR DESCRIPTION
**Pull request by wied03**

### CODE REVIEW

File: library/pathname/relative_path_from_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require 'pathname'
+
+describe "Pathname#relative_path_from" do
+  it "has no common prefix" do
```
Comment (eregon) [inline library/pathname/relative_path_from_spec.rb]: While I am not very strict on it, usually the concatenated description should be a sentence and describe the behavior:
`Pathname#relative_path_from raises an error when the two paths do not share a common prefix`.
It's definitely some work though, but maybe a few here could be improved for readability? (i.e. it also helps to get a quick idea when a spec fails and when reading a spec for the first time)

**Comment by wied03** [comment]
@eregon - Sorry, got tired and sloppy. I revised the descriptions and it also occurred to me to port more of the 'test_unit' specs over for this so I did that. How many Ruby implementations rely on test_unit vs. these specs? Does the MRI team respond quicker to test_unit breakage vs. rubyspec breakage?

**Comment by wied03** [comment]
I did review http://eregon.github.io/rubyspec/2015/07/29/rubyspec-is-reborn.html and that clarified some things. I suppose what I'm really asking is "is there a roadmap for migrating over from test unit tests to rubyspecs"? The write up talks about them being complementary but Opal (to my knowledge) only uses Rubyspecs. Maybe Opal ought to use both?

**Comment by eregon** [comment]
Thanks for the update, looks great!

> is there a roadmap for migrating over from test unit tests to rubyspecs?

Not really, migrating specs is an effort and usually examples from MRI's tests cannot be reused as a simple copy/paste. I find it of course valuable to have more coverage in ruby/spec, but if it tests the exact same code then it has limited value if implementations can run both ruby/spec and MRI tests.
Being able to run MRI tests (e.g.: in Opal) is certainly a valuable effort.

Some other implementations use MRI tests, but only run a fraction of it AFAIK.
From my experience, working with MRI tests tends to be harder since there is no support for automatic tagging and untagging for failures of fine-grained examples, test files are rather large and therefore quickly trigger unsupported features (be it syntax, class-level code, etc) and some tests are implementation-specific.
But some of these tests also have no corresponding specs in ruby/spec, especially for the standard library since these test were usually written by the authors themselves well before RubySpec.

In the specific case of FileUtils, I think it's definitely worth to see if Opal can run the MRI tests.

In general, *I think* contributing to RubySpec is more useful for Ruby implementations at large as ruby/spec strives to provide a spec framework designed for Ruby implementations, from starting to fully-compatible. It also provide descriptions of the behavior which should help implementing features.

**Pull request merged**